### ISSUE TITLE
externalUrl not set in Prometheus and AlertManager definition

### ISSUE THREAD

**Issue by ccjoelm** [coreos/kube-prometheus#77]
I was using the [ingress.jsonnet](https://github.com/coreos/kube-prometheus/blob/master/examples/ingress.jsonnet) file as a starting point for some customizations for our deployment.  In previous releases I had to manually add the externalUrl property to the AlertManager and Prometheus definition as described here [Exposing Prometheus and Alertmanager](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/exposing-prometheus-and-alertmanager.md).  However, if I build my project and review the files generated as manifests/alertmanager-alertmanager.yaml and manifests/prometheus-prometheus.yaml I don't see the externalUrl: 'http://alertmanager.example.com' set.  

I would expect to see the externalUrl parameter populated in those files.  What am I missing?

**Comment by ccjoelm** [comment]
I think it may just be an issue with the [ingress.jsonnet](https://github.com/coreos/kube-prometheus/blob/master/examples/ingress.jsonnet) example.  I tried building using the [minikube.jsonnet](https://github.com/coreos/kube-prometheus/blob/master/examples/minikube.jsonnet) and the externalUrl parameters were populated as expected.

**Comment by brancz** [comment]
Thanks for reporting this. Could you share your jsonnet files, where this doesn't work? :slightly_smiling_face:

**Comment by richerve** [comment]
@ccjoelm the Ingress example doesn't include changes to the spec of either the Prometheus, Alertmanager or Grafana specs. You'll need to specifically add those, for example:

```
    prometheus+:: {
      prometheus+: {
        spec+: {
          externalUrl: "http://prometheus.<your_domain>",
        },
      },
    }
...
```

**Comment by RajalakshmiPrabhakar** [comment]
Hi ALL,
After following the link https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/exposing-prometheus-and-alertmanager.md
able to view the prometheus web page <IP>:30900 but target page is empty.
But able to see grafana metrics and graphs for pods
PFB the nodeport services
teluser@kubemaster:/var/tmp/HPA$ kubectl get services -n monitoring
NAME                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
alertmanager-main       NodePort    10.107.181.153   <none>        9093:30903/TCP      5m16s
alertmanager-operated   ClusterIP   None             <none>        9093/TCP,6783/TCP   5m16s
grafana                 ClusterIP   10.105.47.31     <none>        3000/TCP            66m
grafana-node            NodePort    10.111.178.62    <none>        3000:30902/TCP      5m16s
kube-state-metrics      ClusterIP   None             <none>        8443/TCP,9443/TCP   66m
node-exporter           ClusterIP   None             <none>        9100/TCP            66m
prometheus-adapter      ClusterIP   10.107.86.123    <none>        443/TCP             66m
prometheus-k8s          ClusterIP   10.109.50.252    <none>        9090/TCP            66m
prometheus-main         NodePort    10.100.5.56      <none>        9090:30900/TCP      5m16s
prometheus-operated     ClusterIP   None             <none>        9090/TCP            65m
prometheus-operator     ClusterIP   None             <none>        8080/TCP            66m

**Comment by BenoitKnecht** [comment]
Looks like this has been fixed in f663fb6062c1d345ee7719ad82ec36ed650a943e, I think this issue can be closed.

**Comment by brancz** [comment]
Thanks @BenoitKnecht !
